### PR TITLE
sage.{functions,interfaces,symbolic}: Replace imports from sage.*.all for namespace packages

### LINKED ISSUE #34954: sage.{functions,interfaces,symbolic}: Replace imports from sage.*.all for namespace packages
<div id="comment:0"></div>

Using `./sage -fiximports` from #34945.

Also remove trailing whitespace in the affected files.

Part of Meta-ticket #34201

CC:  @alexchandler100 @dimpase @DaveWitteMorris

Component: **refactoring**

Author: **Alex Chandler, Matthias Koeppe**

_Issue created by migration from https://trac.sagemath.org/ticket/34954_

### PR DESCRIPTION
**Pull request by mkoeppe**
### 📚 Description
Fixes #34954

### 📝 Checklist

- [x] I have made sure that the title is self-explanatory and the description concisely explains the PR.
- [x] I have linked an issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/sage/interfaces/sage0.py [hunk 1]
```diff
@@ -159,7 +159,8 @@ def __init__(self,
         if python:
             command = 'python -u'
             prompt = re.compile(b'>>> ')
-            init_code.append('from sage.all import *')
+            environment = 'sage.all'
```
Comment (tobiasdiez) [inline src/sage/interfaces/sage0.py:162]: This seems like an unrelated and unnecessary change

**Review by tobiasdiez — CHANGES_REQUESTED** [review]
LGTM except for a minor issue.

File: src/sage/interfaces/sage0.py (hunk 1 above)
Comment (mkoeppe) [reply src/sage/interfaces/sage0.py:162]: It is done because otherwise our linter (relint) complains about this line!

Comment (tobiasdiez) [reply src/sage/interfaces/sage0.py:162]: I don't think it's a good idea to change the runtime behavior (although the impact is minimal here) to satisfy a linter. Is there not something like `relint-ignore-nextline` to ignore errors? Or improve the regex to not match in strings... Anyway, this should not be part of this PR...

Comment (mkoeppe) [reply src/sage/interfaces/sage0.py:162]: It is part of this PR because this PR takes care of `sage.interfaces`.
The follow up #35110 turns the relint warnings into errors again, so it has to be fixed.
No, there's no such thing as a "noqa" annotation for relint.

Comment (tobiasdiez) [reply src/sage/interfaces/sage0.py:162]: > The follow up https://github.com/sagemath/sage/pull/35110 turns the relint warnings into errors again, so it has to be fixed.

Then I propose you fix it there.

Comment (mkoeppe) [reply src/sage/interfaces/sage0.py:162]: No thanks.

**Comment by mkoeppe** [comment]
Doctest failure in `sage/interfaces/maxima.py` seems unrelated

**Comment by mkoeppe** [comment]
What needs work here?

File: src/sage/interfaces/sage0.py (hunk 1 above)
Comment (tobiasdiez) [reply src/sage/interfaces/sage0.py:162]: That this instance of an "import" is found by the linter, is a bug in the linter and should be fixed accordingly. Since relint parses complete files, you can add a check that the line before does not contain a magic no-op comment, or fix the bug in another way (e.g. check that the import is not in a string, or at the beginning of a line). There are multiple ways to fix this properly.

**Review by tobiasdiez — CHANGES_REQUESTED** [review]

**Comment by mkoeppe** [comment]
One can always imagine more complicated solutions, but that does not mean that there's anything to fix here.

File: src/sage/interfaces/sage0.py (hunk 1 above)
Comment (dimpase) [reply src/sage/interfaces/sage0.py:162]: We need to progress here one way or another. Is it realistic to fix `relint` so that such a workaround is not needed?

Comment (mkoeppe) [reply src/sage/interfaces/sage0.py:162]: The linting rule is **designed** to flag all hardcoded namespace `.all` imports. No, not just at the beginning of a line, and no, also within strings and within doctests. There's no bug.

**Review by dimpase — APPROVED** [review]
lgtm

File: src/sage/interfaces/sage0.py (hunk 1 above)
Comment (tobiasdiez) [reply src/sage/interfaces/sage0.py:162]: Can you please expand on why this is designed to flag imports in strings? If you look for `.all import.*("|')` then there are only two instances of such imports in strings (four if you include doctests), and both of these are false-positives according to your changes here and in https://github.com/sagemath/sage/pull/35106.

Comment (mkoeppe) [reply src/sage/interfaces/sage0.py:162]: 1) Docstrings are strings.
2) These two instances are not false positives. All imports from `sage.all` need to be updated in some way. Above, in `sage.interfaces.sage0`, this defines a pexpect interface to sage itself; and the use of `sage.all` there will not work in a modularized setting. The solution will be to make the evaluation environment (module) configurable so it becomes suitable for modularized deployments. The change here, introducing a variable `environment` is a meaningful step into this direction; it is along the lines of the modularization changes to the doctester https://github.com/sagemath/sage/blob/develop/src/bin/sage-runtests#L60. 
(The full solution is, of course, out of the scope of this PR.)

Comment (tobiasdiez) [reply src/sage/interfaces/sage0.py:162]: I don't see how hiding from the linter a problematic import that needs a proper fix is a good idea but well...

Comment (mkoeppe) [reply src/sage/interfaces/sage0.py:162]: It is very simple: The linter has made me aware of it, and now I am aware.
